ocw/lib/EC2.py

```diff
@@ -216,6 +218,105 @@ def cleanup_all(self):
+                        rt.delete()
+
+            # delete our endpoints
+            response = self.ec2_client(region).describe_vpc_endpoints(Filters=[{'Name': 'vpc-id', 'Values': [vpcId]}])
+            for ep in response['VpcEndpoints']:
+                if self.dry_run:
+                    self.log_info('Deletion of {} skipped due to dry_run mode', ep)
+                else:
+                    self.log_info('Deleting {}', ep)
+                    self.ec2_client(region).delete_vpc_endpoints(VpcEndpointIds=[ep['VpcEndpointId']])
+            # delete our security groups
+            for sg in vpc.security_groups.all():
+                if sg.group_name != 'default':
+                    if self.dry_run:
+                        self.log_info('Deletion of {} skipped due to dry_run mode', sg)
+                    else:
+                        self.log_info('Deleting {}', sg)
+                        sg.delete()
+            # delete any vpc peering connections
+            response = self.ec2_client(region).describe_vpc_peering_connections(
+                Filters=[{'Name': 'requester-vpc-info.vpc-id', 'Values': [vpcId]}])
+            for vpcpeer in response['VpcPeeringConnections']:
+                vpcpeer_connection = self.ec2_resource(region).VpcPeeringConnection(vpcpeer['VpcPeeringConnectionId'])
+                if self.dry_run:
+                    self.log_info('Deletion of {} skipped due to dry_run mode', vpcpeer_connection)
+                else:
+                    self.log_info('Deleting {}', vpcpeer_connection)
+                    vpcpeer_connection.delete()
+            # delete non-default network acls
+            for netacl in vpc.network_acls.all():
+                if not netacl.is_default:
+                    if self.dry_run:
+                        self.log_info('Deletion of {} skipped due to dry_run mode', netacl)
+                    else:
+                        self.log_info('Deleting {}', netacl)
+                        netacl.delete()
+            # delete network interfaces
+            for subnet in vpc.subnets.all():
+                for interface in subnet.network_interfaces.all():
+                    if self.dry_run:
+                        self.log_info('Deletion of {} skipped due to dry_run mode', interface)
+                    else:
+                        self.log_info('Deleting {}', interface)
+                        interface.delete()
+                if self.dry_run:
+                    self.log_info('Deletion of {} skipped due to dry_run mode', subnet)
+                else:
+                    self.log_info('Deleting {}', subnet)
+                    subnet.delete()
+            if self.dry_run:
+                self.log_info('Deletion of VPC skipped due to dry_run mode')
+            else:
+                # finally, delete the vpc
+                self.ec2_resource(region).meta.client.delete_vpc(VpcId=vpcId)
+        except Exception as e:
```

Comment:
Shouldn't we return `False` in the case a Exception is thrown?

Reply:
yeah I think I need to revert this , this happened because I was already returning `True` when no deletion happened due to running instances attached and I needed to distinguish exception thrown case somehow

Reply:
> yeah I think I need to revert this , this happened because I was already returning `True` when no deletion happened due to running instances attached and I needed to distinguish exception thrown case somehow

ok just ignore this :) I got you wrong first time . yeah I will add `return False` in case of exception catch

Reply:
:+1: